pkg/services/sqlstore/migrations/ualert/ualert.go

```diff
@@ -184,3 +205,22 @@ func (m *migration) Exec(sess *xorm.Session, mg *migrator.Migrator) error {
 
 	return nil
 }
+
+func (m *migration) clear() error {
+	_, err := m.sess.Exec("delete from alert_rule")
```

Comment:
@papagian I think this would still possibly remove folders not created by the migration, e.g.:

- Enable new alerting
- User creates a folder
- User adds rule in folder
- Disable new alerting

or

- User had created an empty folder
- Enable new alerting
- User adds rule in that folder
- Disable new alerting

Maybe an empty folder has some permissions or something else like this, so seems we should be deleting it unless we can be sure it was created for the migration.

Reply:
you are right; I thought that this is fine as long as those folders are only associated with alert rules but it's not.
Also we have to delete folder permissions created during the migration; I think that would be easier as long as we figure out how to spot the folders created during the migration.

Reply:
> I think that would be easier as long as we figure out how to spot the folders created during the migration.

Maybe we can take a specific negative number for the migration to put into `created_by`? May need to check things that already look at that.

Reply:
I like this idea!